File: 667-beautiful-arrangement-ii/ken/667.py
class Solution(object):
    def constructArray(self, n, k):
        """
        :type n: int
        :type k: int
        :rtype: List[int]
        """

        a = []
        for i in xrange(1, n - k):
            a.append(i)

        for i in xrange(k + 1):
            if i % 2 == 0:
                a.append(n - k + i/2)
            else:
                a.append(n - i/2)

        return a

File: 667-beautiful-arrangement-ii/leochang/667.cpp
class Solution {
public:
    vector<int> constructArray(int n, int k) {
        vector<int> ans;
        int begin = 1, end = n;
        for(int i = k; i > 1; i--) {
            if (i % 2 == 0) {
                ans.push_back(end--);
            } else {
                ans.push_back(begin++);
            }
        }
        for(int i = k-1; i < n; i++) {
            ans.push_back(begin++);
        }
        return ans;
    }
};
